src/main/java/jchess/gamelogic/Game.java

```diff
@@ -253,25 +277,29 @@ public void checkStalemate() {
 	public void tryExecuteRule(IMoveCandidate oMoveCandidate) {
 		m_oLogger.writeLog(LogLevel.INFO, "Trying to make move." + oMoveCandidate.toLog(), "tryExecuteRule", "Game");
 
-		
-		
 		deselectedActivePosition();
 		
-//		IPlayerAgent oPlayer = m_oGameState.getActivePlayer();
-////		oPlayer.getAllPieces();
-//		System.out.println(oPlayer.getAllPieces());
-		//check here
-		
 		IBoardActivity oActivity = m_oRuleProcessor.tryExecuteRule(m_oGameModel.getBoard(), oMoveCandidate);
 //		m_oExtendedRuleProcessor.tryEvaluateAllRules(m_oGameModel.getBoard(), oPiece, oMoveCandidate);
 
 		if (oActivity != null) {
 			m_oGameModel.addBoardActivity(oActivity);
+			//Check if the King is still in Check position after the player make the move. Then the current player is lose.
+			if(m_oGameState.getActivePlayer().getName()==m_stPlayerInCheck && tryCheckRule()==true)
+			{
+				JOptionPane.showConfirmDialog(null, "Check Mate!"+"\n"+ m_stPlayerInCheck +"'s King can't escape" +"\n" + m_stPlayerCheck + " is the winner!!!"  , "Winner chicken dinner!", JOptionPane.DEFAULT_OPTION);
+				m_oJChessView.removeCurrentTab();
+				IBoardAgent oNullBoard =  null;
+				m_oGameModel.setBoard(oNullBoard);
+				m_oTimer.stop();
+				
+			}
 			m_oGameState.switchPlayTurn();
 			notifyListenersOnMoveMadeByPlayer(m_oGameState.getActivePlayer(), oActivity);
 			notifyListenersOnCurrentPlayerChanged(m_oGameState.getActivePlayer());
+			//Set the Check time to 0. So the warning would appear again in the next turn
+			m_nCheckTime = 0;
 			
-			//also check here
 		}
 	}
 	
```

Comment:
Kindly move Check rule to KingRuleProcessor class. Call RuleEngine and ask it to check this functionality.